pkg/flow/adapter/transformation/adapter_test.go

```diff
@@ -35,22 +36,29 @@ var availableTransformations = []v1alpha1.Transform{
 	{Operation: "delete"},
 }
 
-func TestNewHandler(t *testing.T) {
-	_, err := newHandler(availableTransformations, availableTransformations)
-	assert.NoError(t, err)
-}
-
 func TestStart(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
With this refactoring, the instantiation of `ctx` and `cancel` ends up being far away from where they are actually used.
For the sake of code clarity, I would recommend moving it right before the place where `a.Start(ctx)` is called.

```diff
+	ctx, cancel := context.WithCancel(context.Background())

	go func() {
		defer close(errChan)
		errChan <- a.Start(ctx)
	}()

	cancel()
```